lib/kino/download.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Kino.Download do
+  @moduledoc false
+
+  use Kino.JS
+  use Kino.JS.Live
+
+  @doc """
+  Renders a download button.  The content of downloaded file is
+  prepared on the server by invoking the `content_fun` argument only
+  when the button is clicked.
+
+  ## Examples
+
+      Kino.Download.new("file.json", fn ->
+        ~s/{"foo": "bar"}/
+      end)
+
+  """
+  def new(filename, content_fun) do
+    Kino.JS.Live.new(__MODULE__, {filename, content_fun})
+  end
+
+  @impl true
+  @spec init({any, any}, Kino.JS.Live.Context.t()) :: {:ok, Kino.JS.Live.Context.t()}
```

Comment:
We don't need typespecs for behaviour callbacks, because those already have a typespec :)

```suggestion
```